pkg/kubernetes/kubernetes_interface.gen.go

```diff
@@ -9,157 +9,232 @@ import (
+	CreateDatabaseClusterRestore(ctx context.Context, restore *everestv1alpha1.DatabaseClusterRestore) (*everestv1alpha1.DatabaseClusterRestore, error)
 	// ListDatabaseEngines returns list of managed database clusters.
-	ListDatabaseEngines(ctx context.Context, namespace string) (*everestv1alpha1.DatabaseEngineList, error)
+	// This method returns a list of full objects (meta and spec).
+	ListDatabaseEngines(ctx context.Context, opts ...ctrlclient.ListOption) (*everestv1alpha1.DatabaseEngineList, error)
 	// GetDatabaseEngine returns database clusters by provided name.
-	GetDatabaseEngine(ctx context.Context, namespace, name string) (*everestv1alpha1.DatabaseEngine, error)
+	GetDatabaseEngine(ctx context.Context, key ctrlclient.ObjectKey) (*everestv1alpha1.DatabaseEngine, error)
 	// UpdateDatabaseEngine updates the provided database engine.
 	UpdateDatabaseEngine(ctx context.Context, engine *everestv1alpha1.DatabaseEngine) (*everestv1alpha1.DatabaseEngine, error)
 	// SetDatabaseEngineLock sets the lock on the database engine.
 	// The lock is automatically set to false once everest-operator completes its upgrade.
 	SetDatabaseEngineLock(ctx context.Context, namespace, name string, locked bool) error
 	// GetDeployment returns k8s deployment by provided name and namespace.
-	GetDeployment(ctx context.Context, name, namespace string) (*appsv1.Deployment, error)
+	GetDeployment(ctx context.Context, key ctrlclient.ObjectKey) (*appsv1.Deployment, error)
 	// UpdateDeployment updates a deployment and returns the updated object.
 	UpdateDeployment(ctx context.Context, deployment *appsv1.Deployment) (*appsv1.Deployment, error)
 	// ListDeployments returns a list of deployments in the provided namespace.
-	ListDeployments(ctx context.Context, namespace string) (*appsv1.DeploymentList, error)
+	// This method returns a list of full objects (meta and spec).
+	ListDeployments(ctx context.Context, opts ...ctrlclient.ListOption) (*appsv1.DeploymentList, error)
 	// DeleteDeployment deletes a deployment by provided name and namespace.
-	DeleteDeployment(ctx context.Context, name, namespace string) error
+	DeleteDeployment(ctx context.Context, obj *appsv1.Deployment) error
+	// RestartDeployment restarts the given deployment.
+	RestartDeployment(ctx context.Context, key ctrlclient.ObjectKey) error
+	// WaitForRollout waits for rollout of a provided deployment in the provided namespace.
+	WaitForRollout(ctx context.Context, key ctrlclient.ObjectKey) error
+	// GetInstallPlan retrieves an OLM install plan by namespace and name.
+	GetInstallPlan(ctx context.Context, key ctrlclient.ObjectKey) (*olmv1alpha1.InstallPlan, error)
+	// UpdateInstallPlan updates OLM install plan.
+	UpdateInstallPlan(ctx context.Context, installPlan *olmv1alpha1.InstallPlan) (*olmv1alpha1.InstallPlan, error)
 	// ApproveInstallPlan approves an install plan.
 	ApproveInstallPlan(ctx context.Context, namespace, installPlanName string) (bool, error)
```

Comment:
Should we use `ctrlclient.Objectkey` here as well?

Reply:
done